lib/resque/worker.rb

```diff
@@ -379,61 +403,30 @@ def unregister_worker
         job.fail(DirtyExit.new)
       end
 
-      redis.srem(:workers, self)
-      redis.del("worker:#{self}")
-      redis.del("worker:#{self}:started")
-
-      Stat.clear("processed:#{self}")
-      Stat.clear("failed:#{self}")
+      redis.pipelined do
+        redis.srem(:workers, self)
+        redis.del("worker:#{self}")
+        redis.del("worker:#{self}:queues")
+      end
     end
 
     # Given a job, tells Redis we're working on it. Useful for seeing
     # what workers are doing and when.
     def working_on(job)
       data = encode \
         :queue   => job.queue,
-        :run_at  => Time.now.strftime("%Y/%m/%d %H:%M:%S %Z"),
+        :run_at  => Time.now.utc.iso8601,
```

Comment:
This change is included in master [here](https://github.com/al2o3cr/resque/pull/2/files#diff-693c73ae6c0943d273cdb8efc95ecec2R704)